### PR TITLE
fix(claim-dapp): use reach ui for modal

### PR DESCRIPTION
**Pull request by Gamaranto**
**Motivation**

This PR fixes a bug where it was not possible to scroll the modal when it overflowed the page.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

Signed-off-by: Gamaranto <francesco@umaproject.org>

### CODE REVIEW

File: src/components/Modal.tsx [hunk 1]
```diff
@@ -9,38 +10,50 @@ type ModalProps = {
...
+//         <ExitButton onClick={onClose}>
+//           <Times />
+//         </ExitButton>
+//         <Content>{children}</Content>
+//       </DialogOverlay>
+//       <BgBlur />
+//     </Portal>
+//   );
+// };
+//const Modal = React.forwardRef(ModalComponent);
```
Comment (daywiss) [inline src/components/Modal.tsx:32]: can we remove this?

File: src/components/Modal.tsx [hunk 2]
```diff
@@ -9,38 +10,50 @@ type ModalProps = {
...
       </Wrapper>
-      <BgBlur />
-    </Portal>
+    </Overlay>
   );
 };
 
-const Modal = React.forwardRef(ModalComponent);
-Modal.displayName = "Modal";
+//Modal.displayName = "Modal";
```
Comment (daywiss) [inline src/components/Modal.tsx:47]: should this be put in

**Review by daywiss — COMMENTED** [review]
i dont really know how to test this. it would be great if you could describe what you did to fix this and a screenshot or gif to illustrate the change

File: src/components/Modal.tsx (hunk 1 above)
Comment (Gamaranto) [reply src/components/Modal.tsx:32]: Yup sorry! Nice catch

File: src/components/Modal.tsx (hunk 2 above)
Comment (Gamaranto) [reply src/components/Modal.tsx:47]: Sorry! Nice catch

**Comment by Gamaranto** [comment]
> i dont really know how to test this. it would be great if you could describe what you did to fix this and a screenshot or gif to illustrate the change

Before the changes, on small screens, the modal could overflow the page and scrolling would be prevented:
CleanShot 2021-06-17 at 20 18 14@2x

Now, after the changes, if the modal overflows, you can scroll it! (also notice how the blur of the overlay covers the whole background, while before it stopped when the modal overflowed)

CleanShot 2021-06-17 at 20 21 29

Code wise, I've removed our custom implementation for the `@reach/ui` one, which is battle tested, accessible, and well maintained.

**Review by daywiss — APPROVED** [review]
looks good

**Pull request merged**